src/resolve/resolved-conf.c

```diff
@@ -599,3 +599,37 @@ int manager_parse_config_file(Manager *m) {
         return 0;
 
 }
+
+int config_parse_dns_cache_size(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Manager *m = ASSERT_PTR(userdata);
+        int ret, val;
+
+        assert(filename);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                m->cache_size = CACHE_MAX_DEFAULT;
+                return 0;
+        }
+
+        ret = safe_atoi(rvalue, &val);
```

Comment:
safe_atoi() parses a *signed* integer. That's makes no sense here, what is a negative cache size supposed to be?

Reply:
You're correct, I updated it to a uint64, there willever be a negative value.